optimum/habana/accelerate/utils/transformer_engine.py

```diff
@@ -0,0 +1,137 @@
+# coding=utf-8
+# Copyright 2024 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import torch
+from peft.tuners.lora.layer import Linear as PEFTLinear
+from optimum.habana.peft.layer import LoRALinear
+
+try:
+    import habana_frameworks.torch.hpex.experimental.transformer_engine as te
+    from habana_frameworks.torch.hpex.experimental.transformer_engine.distributed import activation_checkpointing
+
+    has_transformer_engine = True
+except ImportError:
+    has_transformer_engine = False
+
+
+def is_fp8_available():
+    return has_transformer_engine
+
+
+def _convert_model(model, to_transformer_engine=True, _convert_linear=True):
+    """
+    Recursively converts the linear and layernorm layers of a model to their `transformers_engine` counterpart.
+    """
+    if not is_fp8_available():
+        raise ImportError("Using `convert_model` requires transformer_engine to be installed.")
+    for name, module in model.named_children():
+        if type(module) == PEFTLinear and to_transformer_engine and _convert_linear:
+            LoRALinear.replace_forward(module)
+        if isinstance(module, torch.nn.Linear) and not type(module) == PEFTLinear and to_transformer_engine and _convert_linear:
```

Comment:
Add condition to skip LoRA-A LoRA-B from TE wrapping. Does not give much performance gain and there is a chance of accuracy degrade.